### PR TITLE
Publish multiselected Files to Gist

### LINKED ISSUE #851: Publish to GIST with a right click on multi selected files
With multiple files selected, it would be great - with a RT click to have the option to publish to a GIST.

### PR DESCRIPTION
**Pull request by joeizang**
fixes #851

### CODE REVIEW

File: libs/remix-ui/workspace/src/lib/actions/index.ts [hunk 1]
```diff
@@ -274,6 +318,7 @@ export const publishToGist = async (path?: string) => {
       id = await plugin.call('filePanel', 'isGist')
     }
     const packaged = await packageGistFiles(folder)
+    console.log('packed files', packaged)
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/actions/index.ts:321]: remove log

File: libs/remix-ui/workspace/src/lib/actions/index.ts [hunk 2]
```diff
@@ -325,9 +370,12 @@ export const publishToGist = async (path?: string) => {
           public: true,
           files: packaged
         }, (error, result) => {
+          console.log('what comes back from gist creation', { error, result })
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/actions/index.ts:373]: here too

File: libs/remix-ui/workspace/src/lib/actions/index.ts [hunk 3]
```diff
@@ -537,6 +585,7 @@ export const restoreBackupZip = async () => {
 const packageGistFiles = async (directory) => {
   const workspaceProvider = plugin.fileProviders.workspace
   const isFile = await workspaceProvider.isFile(directory)
+  console.log('here is the state of things in packageGistFiles', { directory, isFile })
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/actions/index.ts:588]: here

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx [hunk 4]
```diff
@@ -150,6 +152,38 @@ export function Workspace() {
...
+   */
+  const handlePublishingMultiSelectedFilesToGist = async () => {
+    try {
+      const selectedFiles = []
+      for (const one of feTarget) {
+        if (one.type === 'folder') return
+        const content = await global.plugin.call('fileManager', 'readFile', one.key)
+        selectedFiles.push({ key: one.key, type: one.type, content: content })
+      }
+      console.log('Files selected', selectedFiles)
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:173]: this one

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx [hunk 5]
```diff
@@ -150,6 +152,38 @@ export function Workspace() {
...
+      global.dispatchPublishFilesToGist(selectedFiles)
+    } catch (error) {
+      await global.plugin.call('notification', 'toast', 'Could not publish files to gist. There was an error')
+      await global.plugin.call('notification', 'toast', typeof(error) === 'string' ? error : `${console.log(error)} check the console for more details`)
+    }
+  }
+
+  useEffect(() => {
+    global.plugin.on('finishedGistPublish', (folderName) => {
+      console.log('finished publishing to gist', folderName)
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:183]: here

**Review by Aniket-Engg — COMMENTED** [review]
Functionality wise working fine

File: libs/remix-ui/workspace/src/lib/actions/index.ts (hunk 3 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/actions/index.ts:588]: done

File: libs/remix-ui/workspace/src/lib/actions/index.ts (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/actions/index.ts:373]: done

File: libs/remix-ui/workspace/src/lib/actions/index.ts (hunk 1 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/actions/index.ts:321]: done

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx (hunk 5 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:183]: done

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx (hunk 4 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:173]: done

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**